src/include/kdbmacros.h

```diff
@@ -37,24 +37,33 @@
 	do                                                                                                                                 \
 	{                                                                                                                                  \
 		if (errno == EACCES)                                                                                                       \
-			ELEKTRA_SET_ERROR (109, parentKey, strerror (errno));                                                              \
+			ELEKTRA_SET_RESOURCE_ERRORF (                                                                                      \
+				parentKey,                                                                                                 \
+				"Insufficient permissions to open configuration file for reading. Reason: %s. You might want "             \
```

Comment:
for reading what?

Reply:
What is unclear here? I think it is a decent error message. This macro was implemented by you 3 years ago, I guess you know it better what should be here